### PR TITLE
Add practice exercise `phone-number`

### PR DESCRIPTION
**Pull request by jiegillet**
This one is great for practicing `Result` chains.

### CODE REVIEW

File: exercises/practice/phone-number/test/phone_number_test.gleam [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import gleeunit/should
+import phone_number
+
+pub fn main() {
+  gleeunit.main()
+}
+
+pub fn cleans_the_number_test() {
+  phone_number.clean("(223) 456-7890")
+  |> should.equal(Ok("2234567890"))
```
Comment (jiegillet) [inline exercises/practice/phone-number/test/phone_number_test.gleam:11]: I tried the `assert Ok("..") = ` versus the `|> should.equal(Ok(".."))` patterns, and I found that the error messages were not equally helpful. 
The `assert` pattern gave me
```
  1) phone_number_test.cleans_numbers_with_dots_test
     Assertion pattern match failed
        value: Error("punctuations not permitted")
     location: phone_number_test.cleans_numbers_with_dots_test:14
     stacktrace:
       phone_number_test.cleans_numbers_with_dots_test
     output: 
```
while the `should` pattern gave me
```
  1) phone_number_test.cleans_numbers_with_dots_test
     Values were not equal
     expected: Ok("2234567890")
          got: Error("punctuations not permitted")
     output: 
```

While the first one game me a location, the second one gave me the expectation directly, which in this case was more helpful, so I went with that.
Once the test runner implements parsing the snippets of code, the situation might be different

Comment (lpil) [reply exercises/practice/phone-number/test/phone_number_test.gleam:11]: I think in this case it is good either way, though `assert` will continue to improve and we'll remove `should` in future.

File: exercises/practice/phone-number/.meta/example.gleam [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import gleam/string
+import gleam/result
+import gleam/list
+
+pub fn clean(input: String) -> Result(String, String) {
+  Ok(input)
+  |> result.map(remove_valid_punctuation)
+  |> result.then(validate_characters)
```
Comment (lpil) [inline exercises/practice/phone-number/.meta/example.gleam:8]: The first bit of wrapping can be skipped
```suggestion
  input
  |> remove_valid_punctuation
  |> validate_characters
```

**Review by lpil — APPROVED** [review]
Nice!

File: exercises/practice/phone-number/test/phone_number_test.gleam (hunk 1 above)
Comment (jiegillet) [reply exercises/practice/phone-number/test/phone_number_test.gleam:11]: Good to know!

**Pull request merged**